pkg/appgateserver/config/appgate_configs_reader_test.go

```diff
@@ -0,0 +1,115 @@
+package config_test
+
+import (
+	"net/url"
+	"testing"
+
+	sdkerrors "cosmossdk.io/errors"
+	"github.com/gogo/status"
+	"github.com/stretchr/testify/require"
+
+	"github.com/pokt-network/poktroll/pkg/appgateserver/config"
+	"github.com/pokt-network/poktroll/testutil/yaml"
+)
+
+func Test_ParseAppGateConfigs(t *testing.T) {
+	tests := []struct {
+		desc     string
+		err      *sdkerrors.Error
+		expected *config.AppGateConfig
+		config   string
+	}{
+		// Valid Configs
+		{
+			desc: "appgate_config_test: valid app gate config",
```

Comment:
NIT (leaving just one more comment). Decide on `app gate` vs `appgate` and be consistent in comments and strings